mbed_greentea/mbed_greentea_cli.py

```diff
@@ -837,13 +837,18 @@ def main_cli(opts, args, gt_instance_uuid=None):
     for t in execute_threads:
         t.daemon = True
         t.start()
-    while test_result_queue.qsize() != len(execute_threads):
-        sleep(1)
 
     # merge partial test reports from diffrent threads to final test report
     for t in execute_threads:
-        t.join()
-        test_return_data = test_result_queue.get(False)
+        try:
+            t.join() #blocking
+            test_return_data = test_result_queue.get(False)
+        except Exception as e:
+            # No test report generated
+            gt_logger.gt_log_err("could not generate test report")
```

Comment:
Because you are taking Exception name 'e' you can print it as well ! :)

```python
gt_logger.gt_log_err("could not generate test report: " + str(e))
```